pkg/portutil/procnet/procnet.go

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package procnet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+type NetworkDetail struct {
+	LocalIP   net.IP
+	LocalPort uint64
+}
+
+func Parse(data []string) (results []NetworkDetail) {
+	temp := removeEmpty(data)
+	for _, value := range temp {
+		lineData := removeEmpty(strings.Split(strings.TrimSpace(value), " "))
+		ip, port, err := ParseAddress(lineData[1])
+		if err != nil {
+			continue
+		}
+		results = append(results, NetworkDetail{
+			LocalIP:   ip,
+			LocalPort: uint64(port),
+		})
+	}
+	return results
+}
+
+func removeEmpty(array []string) (results []string) {
+	for _, i := range array {
+		if i != "" {
+			results = append(results, i)
+		}
+	}
+	return results
+}
+
+func ParseAddress(s string) (net.IP, uint16, error) {
```

Comment:
Please clarify the origin of this code and its license in the comment lines
https://github.com/lima-vm/lima/blob/v0.8.3/pkg/guestagent/procnettcp/procnettcp.go#L95-L137